taurus/json/taurus_json.py

```diff
@@ -0,0 +1,322 @@
+import inspect
+
+from taurus.entity.attribute.condition import Condition
+from taurus.entity.attribute.parameter import Parameter
+from taurus.entity.attribute.property import Property
+from taurus.entity.attribute.property_and_conditions import PropertyAndConditions
+from taurus.entity.base_entity import BaseEntity
+from taurus.entity.bounds.categorical_bounds import CategoricalBounds
+from taurus.entity.bounds.composition_bounds import CompositionBounds
+from taurus.entity.bounds.integer_bounds import IntegerBounds
+from taurus.entity.bounds.real_bounds import RealBounds
+from taurus.entity.file_link import FileLink
+from taurus.entity.link_by_uid import LinkByUID
+from taurus.entity.object import ProcessRun, MaterialRun, MeasurementRun
+from taurus.entity.object.ingredient_run import IngredientRun
+from taurus.entity.object.ingredient_spec import IngredientSpec
+from taurus.entity.object.material_spec import MaterialSpec
+from taurus.entity.object.measurement_spec import MeasurementSpec
+from taurus.entity.object.process_spec import ProcessSpec
+from taurus.entity.source.performed_source import PerformedSource
+from taurus.entity.template.condition_template import ConditionTemplate
+from taurus.entity.template.material_template import MaterialTemplate
+from taurus.entity.template.measurement_template import MeasurementTemplate
+from taurus.entity.template.parameter_template import ParameterTemplate
+from taurus.entity.template.process_template import ProcessTemplate
+from taurus.entity.template.property_template import PropertyTemplate
+from taurus.entity.value.discrete_categorical import DiscreteCategorical
+from taurus.entity.value.empirical_formula import EmpiricalFormula
+from taurus.entity.value.nominal_categorical import NominalCategorical
+from taurus.entity.value.nominal_composition import NominalComposition
+from taurus.entity.value.nominal_integer import NominalInteger
+from taurus.entity.value.nominal_real import NominalReal
+from taurus.entity.value.normal_real import NormalReal
+from taurus.entity.value.uniform_integer import UniformInteger
+from taurus.entity.value.uniform_real import UniformReal
```

Comment:
For the scale of this list and the nature of the import, wouldn't it make more sense to `import *`?  If those imports auto-registered the modules with _clazzes it then we remove fragility.

Reply:
> For the scale of this list and the nature of the import, wouldn't it make more sense to `import *`?

Sure, but we don't have the imports to support that: https://github.com/CitrineInformatics/taurus/blob/develop/taurus/entity/value/__init__.py

I'd like to treat that as a separate issue.

> If those imports auto-registered the modules with _clazzes it then we remove fragility.

I'm not sure what you mean here.  Are you suggesting reflecting into the members of a module?

Reply:
Not entire sure if it's textbook reflection, but basically yes.  

There are a few ways to do it, but it would mostly make sense if taurus/json were tied strongly to DictSerializable, as opposed to being an external module that crawls its trees.

Reply:
Could you give an example of a way to do it?  If this is merged by then, then you could do it in another PR.  Seems like a relatively localized change.